sdks/java/core/src/main/java/org/apache/beam/sdk/io/TextSource.java

```diff
@@ -116,8 +118,14 @@ static class TextBasedReader extends FileBasedReader<String> {
 
     private final byte @Nullable [] delimiter;
     private final int skipHeaderLines;
-    private final ByteArrayOutputStream str;
+
+    // The output stream can contain the delimiter at the last. It must exclude the delimiter when
```

Comment:
How about replacing the first sentence nd maybe add something about how this is used:
"Used to build up results that span buffers. It may contain the delimiter as a suffix."

Reply:
Done